lightningd/invoice.c

```diff
@@ -267,6 +268,7 @@ void invoice_try_pay(struct lightningd *ld,
 
 	log_debug(ld->log, "Calling hook for invoice '%s'", details->label->s);
 	plugin_hook_call_invoice_payment(ld, payload, payload);
+	notify_invoice_payment(ld, payload->msat, payload->preimage, payload->label);
```

Comment:
At this point we no longer have ownership over the payload, so maybe it's better to switch the two calls, or to create a copy.